lib/IRGen/GenDecl.cpp

```diff
@@ -644,6 +665,42 @@ void IRGenModule::addProtocolConformanceRecord(
   ProtocolConformances.push_back(conformance);
 }
 
+static bool
+typeHasExplicitProtocolConformance(CanType type) {
+  auto conformances =
+    type->getAnyNominal()->getAllConformances();
```

Comment:
Fixed, I think we want to return true (non-nominal types should not have a type metadata record). Arguably `typeHasExplicitProtocolConformance()` should be renamed to something like `typeNeedsTypeMetadataRecord()`.